ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml

```diff
@@ -342,6 +391,181 @@ objects:
+          SURVIVING_MASTER_MACHINE=$(oc get machine -l machine.openshift.io/cluster-api-machine-role=master -n openshift-machine-api -o name | cut -d '/' -f 2)
+          SURVIVING_MASTER_NUM=${SURVIVING_MASTER_MACHINE##*-}
+          SURVIVING_MASTER_PREFIX=${SURVIVING_MASTER_MACHINE%-*}
+          retry 10 oc get --export machine ${SURVIVING_MASTER_MACHINE} -n openshift-machine-api -o yaml > /tmp/machine.template
+          SURVIVING_MASTER_AZ=$(grep -oP 'availabilityZone: (\K.+)' /tmp/machine.template)
+          SURVIVING_MASTER_AZ_INDEX=$(grep -oP 'availabilityZone: .*(\K.)' /tmp/machine.template)
+
+          MASTER_INDEX=0
+          for i in $(seq 0 1); do
+            if [[ "${MASTER_INDEX}" == "${SURVIVING_MASTER_NUM}" ]]; then MASTER_INDEX=$((MASTER_INDEX+1)); fi
+            cat /tmp/machine.template \
+              | sed 's;selfLink.*;;g' \
+              | sed "s;name: ${SURVIVING_MASTER_PREFIX}-${SURVIVING_MASTER_NUM};name: ${SURVIVING_MASTER_PREFIX}-${MASTER_INDEX};" > /tmp/machine_${i}.yaml
+              retry 10 oc create -n openshift-machine-api -f /tmp/machine_${i}.yaml
+              MASTER_INDEX=$((MASTER_INDEX+1))
+          done
+
+          echo "Waiting for machines to be created"
+          NEW_MASTER_IPS=()
+          for i in $(seq 0 60); do
+            NEW_MASTER_IPS=($(oc -n openshift-machine-api \
+              get machines \
+              -l machine.openshift.io/cluster-api-machine-role=master \
+              -o jsonpath='{range .items[*]}{.status.addresses[?(@.type=="InternalIP")].address}{"\n"}{end}' || true))
+            if [[ "${#NEW_MASTER_IPS[@]}" == "3" ]]; then break; fi
+            sleep 30
+          done
+          oc get machines -n openshift-machine-api
+          if [[ "${#NEW_MASTER_IPS[@]}" != "3" ]]; then
+            echo "${NEW_MASTER_IPS[@]}"
+            exit 1
+          fi
+
+          echo "Verify new master nodes have joined the cluster"
+          FOUND_MASTERS=0
+          for i in $(seq 1 60)
+          do
+            FOUND_MASTERS=($(oc --request-timeout=5s get nodes -l node-role.kubernetes.io/master= -o name --no-headers || true))
+            if [[ "${#FOUND_MASTERS[@]}" == "3" ]]; then break; fi
+            sleep 30
+          done
+          oc get nodes
+          if [[ "${#FOUND_MASTERS[@]}" != "3" ]]; then
+            echo "${FOUND_MASTERS[@]}"
+            exit 1
+          fi
+
+          echo "Update DNS and LB"
+          # aws cli magic
+          easy_install --user pip
+          ~/.local/bin/pip install --user boto3
+          cat > /tmp/update_route_53.py <<'PYTHON_EOF'
+        import boto3
+        import os
+        import sys
+
+        if len(sys.argv) < 3:
+            print("Usage: ./update_route_53.py <RECORD> <IP>")
+            sys.exit(1)
+
+        record = sys.argv[1]
+        ip = sys.argv[2]
+        print("record: %s" % record)
+        print("ip: %s" % ip)
+
+        domain = "%s.%s" % (os.environ["CLUSTER_NAME"], os.environ["BASE_DOMAIN"])
```

Comment:
https://github.com/openshift/installer/blob/f333d684f6553fcaf4f27e263794905da5abc6e2/data/data/bootstrap/files/usr/local/bin/installer-gather.sh#L104

Why not used this?

Reply:
Fixed by passing `DOMAIN` there, extracted from config via `oc`